### PR TITLE
Consensus test tweaks

### PR DESCRIPTION
**Pull request by teor2345**
Fixes minor issues identified in PRs #492 and #517, and fixes the tracing bug in #515 for block and checkpoint in zebra-consensus.

### CODE REVIEW

**Review by dconnolly — DISMISSED** [review]
Lgtm!

File: zebra-consensus/src/verify/block.rs [hunk 1]
```diff
@@ -171,4 +289,186 @@ mod tests {
...
+
+        let block =
+            Arc::<Block>::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_415000_BYTES[..])?;
+        let hash: BlockHeaderHash = block.as_ref().into();
+
+        let mut state_service = zebra_state::in_memory::init();
+        let mut block_verifier = super::init(state_service.clone());
+
+        /// Make sure the verifier service is ready (1/2)
+        let ready_verifier_service = block_verifier.ready_and().await.map_err(|e| eyre!(e))?;
```
Comment (yaahc) [inline zebra-consensus/src/verify/block.rs:373]: Oh, I see. The double await issue presents an issue here :( I'll start brain storming ways to support this better...

Comment (teor2345) [reply zebra-consensus/src/verify/block.rs:373]: A similar issue also turns up if you try to add a span to a parent block containing multiple awaits. Even if each await already has its own span. (For example, a loop containing multiple awaits.)

**Review by yaahc — APPROVED** [review]

File: zebra-consensus/src/verify/block.rs (hunk 1 above)
Comment (yaahc) [reply zebra-consensus/src/verify/block.rs:373]: aah yea, i figured that would happen based on the way I implemented it the actual instrumentation ontop of syn's expression visitor, it just iterates over statements within every block and for each statement that has a doc comment it tries to instrument the await with an await visitor, but it cant tell the difference between an await thats in a statement inside of a block inside of the current statement or if its an await thats directly part of the expression you instrumented. Not really sure how to fix it... maybe @mystor would have some ideas

**Pull request merged**